django/db/backends/sqlite3/base.py

```diff
@@ -360,16 +360,21 @@ def _sqlite_date_extract(lookup_type, dt):
 
 
 def _sqlite_date_trunc(lookup_type, dt):
-    try:
-        dt = backend_utils.typecast_timestamp(dt)
-    except (ValueError, TypeError):
-        return None
-    if lookup_type == 'year':
-        return "%i-01-01" % dt.year
-    elif lookup_type == 'month':
-        return "%i-%02i-01" % (dt.year, dt.month)
-    elif lookup_type == 'day':
-        return "%i-%02i-%02i" % (dt.year, dt.month, dt.day)
+    try: 
+        try:
+            dt = backend_utils.typecast_timestamp(dt)
+        except (ValueError, TypeError):
+            return None
+        if dt is None: 
+            return None
+        if lookup_type == 'year':
+            return "%i-01-01" % dt.year
+        elif lookup_type == 'month':
+            return "%i-%02i-01" % (dt.year, dt.month)
+        elif lookup_type == 'day':
+            return "%i-%02i-%02i" % (dt.year, dt.month, dt.day)
+    except Exception as e: 
```

Comment:
Could you remove this outer try/except (it was only for debugging)? The build is failing because `print` needs to be `print(e)` for Python 3 compatibility but that can be removed anyway.